components/RichTextEditor.js

```diff
@@ -273,6 +283,107 @@ export default class RichTextEditor extends React.Component {
+    const videoEmbedDialog = `
+            <div class="trix-dialog" data-trix-dialog="video-url" data-trix-dialog-attribute="video">
+              <div class="trix-dialog__link-fields">
+                <input type="url" name="video-url" class="trix-input trix-input--dialog trix-input--dialog-video" placeholder="Enter Video URL…" aria-label="Video URL" data-trix-input="">
+                <div class="trix-button-group">
+                  <input type="button" class="trix-button trix-button--dialog" value="Add Video" data-trix-action="x-video-embed">
+                </div>
+              </div>
+              <strong>Note: Only YouTube and Vimeo links are supported.</strong>
+            </div>`;
+    const { toolbarElement } = e.target;
+    const attachFilesButton = toolbarElement.querySelector('[data-trix-action=attachFiles]');
+    attachFilesButton.insertAdjacentHTML('afterend', videoEmbedButton);
+    const trixDialog = toolbarElement.querySelector('.trix-dialog--link');
+    trixDialog.insertAdjacentHTML('afterend', videoEmbedDialog);
+  };
+
+  trixActionInvoke = e => {
+    const { toolbarElement } = e.target;
+    if (e.actionName === 'x-video-dialog-open') {
+      const attachVideoDialog = toolbarElement.querySelector('[data-trix-dialog=video-url]');
+      const attachVideoDialogInput = toolbarElement.querySelector('.trix-input--dialog-video');
+      if (attachVideoDialog.getAttribute('data-trix-active') === '') {
+        attachVideoDialog.removeAttribute('data-trix-active');
+      } else {
+        attachVideoDialog.setAttribute('data-trix-active', '');
+        attachVideoDialogInput.removeAttribute('disabled');
+      }
+    } else if (e.actionName === 'x-video-embed') {
+      const videoLink = toolbarElement.querySelector('.trix-input--dialog-video').value;
+      if (videoLink) {
+        this.embedVideoIFrame(videoLink);
+      }
+    }
+  };
+
+  constructVideoEmbedURL = videoLink => {
+    const { id, service } = this.getVideoId(videoLink);
+    if (service === 'youtube') {
+      return `https://www.youtube.com/embed/${id}`;
+    } else if (service === 'vimeo') {
+      return `https://player.vimeo.com/video/${id}`;
+    } else {
+      return null;
+    }
+  };
+
+  getVideoId = videoLink => {
+    let regexExp;
+    const videoId = {};
+    if (videoLink.includes('youtu')) {
+      videoId.service = 'youtube';
+      regexExp = new RegExp(
+        '(?:https?://)?(?:www.)?youtu(?:.be/|be.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
+        'ig',
+      );
+    } else if (videoLink.includes('vimeo')) {
+      videoId.service = 'vimeo';
+      regexExp = new RegExp(
+        '(http|https)?://(www.)?vimeo.com/(?:channels/(?:\\w+/)?|groups/([^/]*)/videos/|)(\\d+)(?:|/?)',
+      );
+    } else {
+      return videoId;
+    }
+    const matches = regexExp.exec(videoLink);
+    if (matches) {
+      videoId.id = matches[matches.length - 1];
+    }
+    return videoId;
+  };
```

Comment:
`videoLink.includes('youtu')` is not a great way to check for this. If the vimeo channel has `youtu` in its name, it will be fooled by this check.

Also it's strange to return a `videoId` that is in fact an object, not an id. Suggestion:

```suggestion
  parseVideoLink = videoLink => {
      const regexps = {
        youtube: new RegExp(
          '(?:https?://)?(?:www.)?youtu(?:.be/|be.com/\\S*(?:watch|embed)(?:(?:(?=/[^&\\s?]+(?!\\S))/)|(?:\\S*v=|v/)))([^&\\s?]+)',
          'ig',
        ),
        vimeo: new RegExp(
          '(http|https)?://(www.)?vimeo.com/(?:channels/(?:\\w+/)?|groups/([^/]*)/videos/|)(\\d+)(?:|/?)',
        ),
      };

      for (const service in regexps) {
        const matches = regexps[service].exec(videoLink);
        if (matches) {
          return { service, id: matches[matches.length - 1] };
        }
      }

      return {};
  };
```

Reply:
@Betree : Thanks for the suggestion. I've changed this. 👍🏽